### DISCUSSION TITLE
Not understanding the function of --proxy-domain

### DISCUSSION THREAD

**Discussion by edwardwoolls** [coder/code-server#6110] [Q&A]
I currently have code-server set up and running on Centos 9, but I am having trouble understanding the use of the "--proxy-domain" option. I've setup up my Caddyfile to display "localhost:3000" on "3000.domain.tld", which works even if I run code-server without the "--proxy-domain" option. That led me to think it would use port.domain.tld when I run dev on code-server, but it doesn't. It still goes to "domain.tld/proxy/3000". Can someone please explain this to me? I may be using the option incorrectly or missing a step. At this point, I am considering redirecting "/proxy/xxx" to "xxx.domain.tld" instead.

**Comment by code-asher** [comment]
Are you talking about the ports panel?  Unfortunately that still defaults to the path-based proxy.  We need to make it use the domain-based proxy when `--proxy-domain` is set (https://github.com/coder/code-server/issues/5311).

It is possible to force the panel to use the right proxy by setting `VSCODE_PROXY_URI`.  For example:

```
docker run --rm -e 'VSCODE_PROXY_URI=https://{{port}}.domain.tld' -p 8080:8080 codercom/code-server --proxy-domain domain.tld
```

Then point `domain.tld` and `*.domain.tld` to code-server via Caddy.
